src/server/db_slice.cc

```diff
@@ -1129,16 +1124,14 @@ void DbSlice::FlushChangeToEarlierCallbacks(DbIndex db_ind, Iterator it, uint64_
   }
 }
 
-//! Unregisters the callback.
-void DbSlice::UnregisterOnChange(uint64_t id) {
-  lock_guard lk(cb_mu_);  // we need to wait until callback is finished before remove it
-  for (auto it = change_cb_.begin(); it != change_cb_.end(); ++it) {
-    if (it->first == id) {
-      change_cb_.erase(it);
-      return;
-    }
-  }
-  LOG(DFATAL) << "Could not find " << id << " to unregister";
+void DbSlice::CallOnChange(DbIndex id, const ChangeReq& cr) const {
+  FiberAtomicGuard fg;  // Callbacks don't preemept
```

Comment:
preempt

Reply:
but soon they will preempt. is there any problem with this design have the callbacks preempting?